cmd/state-installer/installer.go

```diff
@@ -83,6 +80,11 @@ func (i *Installer) Install() (rerr error) {
 		return locale.WrapInputError(err, "err_update_corrupt_install", constants.DocumentationURL)
 	}
 
+	err = legacytray.DetectAndRemove(i.path, i.cfg)
+	if err != nil {
+		multilog.Error("Unable to detect and/or remove legacy tray. Will try again next update. Error: %v", err)
```

Comment:
There is a question of what exactly to do in this case. If the user attempts to uninstall the State Tool after a failed removal here, then state-tray will remain. On the other hand, if we error out during the update, that might cause them trouble.

Reply:
Could we just add this to the uninstaller as well? And have that one error out if it fails.

Reply:
It got a bit messy, but I've moved the legacytray package into a shared area and use it from both the installer and uninstaller.